plutus-core-spec/figures/PlutusCoreTypeCorrectness.tex

```diff
@@ -132,72 +134,77 @@
-    
+
     \begin{prooftree}
-        \AxiomC{\(\hypJ{\Gamma}{\istermJ{L}{\funT{T}{S}}}\)}
-        \AxiomC{\(\hypJ{\Gamma}{\istermJ{M}{T}}\)}
-        \RightLabel{app}
-        \BinaryInfC{\(\hypJ{\Gamma}{\istermJ{\app{L}{M}}{S}}\)}
+        \AxiomC{\(\hypJ{\Gamma}{\istermJ{M}{\fixT{\alpha}{T}}}\)}
+        \RightLabel{unwrap}
+        \UnaryInfC{\(\hypJ{\Gamma}{\istermJ{\unwrap{M}}{\normalform{\subst{\fixT{\alpha}{T}}{\alpha}{T}}}}\)}
     \end{prooftree}
-    
+
     \begin{prooftree}
-        \AxiomC{$bi$ has type $S$ in Fig. \ref{fig:Plutus_core_builtins}}
-        \RightLabel{builtin}
-        \UnaryInfC{\(\hypJ{\Gamma}{\istermJ{\con{cn}}{S}}\)}
+        \AxiomC{\(\hypJ{\Gamma}{\istypeJ{A}{\typeK{}}}\)}
+        \AxiomC{\(\hypJ{\Gamma, \termJ{y}{\normalform{A}}}{\istermJ{M}{T}}\)}
+        \RightLabel{lam}
+        \BinaryInfC{\(\hypJ{\Gamma}{\istermJ{\lam{y}{A}{M}}{\funT{\normalform{A}}{T}}}\)}
     \end{prooftree}
-    
+
     \begin{prooftree}
-        \AxiomC{\(\hypJ{\Gamma}{\istypeJ{A}{\typeK{}}}\)}
-        \AxiomC{\(\typeMultistep{A}{S}\)}
-        \RightLabel{error}
-        \BinaryInfC{\(\hypJ{\Gamma}{\istermJ{\error{A}}{S}}\)}
+        \AxiomC{\(\hypJ{\Gamma}{\istermJ{L}{\funT{S}{T}}}\)}
+        \AxiomC{\(\hypJ{\Gamma}{\istermJ{M}{S}}\)}
+        \RightLabel{app}
+        \BinaryInfC{\(\hypJ{\Gamma}{\istermJ{\app{L}{M}}{T}}\)}
     \end{prooftree}
 
     \begin{prooftree}
-        \AxiomC{$\Gamma \vdash Q :: \typeK$}
-        \AxiomC{$Q = \mathcal{E}\{\fixT{\alpha}{S}\}$}
-        \AxiomC{$\Gamma \vdash M : Q$}
-        \RightLabel{unwrap}
-        \TrinaryInfC{\(\hypJ{\Gamma}{\istermJ{\unwrap{M}}{[\fixT{\alpha}{S}}/\alpha]S}\)}
+        \alwaysNoLine
+        \AxiomC{$bn$ has signature $\sig{\alpha_0 :: K_0, ..., \alpha_m :: K_m}{B_0, ..., B_n}{C}$ in Fig. \ref{fig:Plutus_core_builtins}}
+        \UnaryInfC{\(\hypJ{\Gamma}{\istermJ{M_i}{\normalform{\subst{A_0, ..., A_n}{\alpha_0, ..., \alpha_m}{B_i}}}}\)}
+        \alwaysSingleLine
+        \RightLabel{builtin}
+        \UnaryInfC{\(\hypJ{\Gamma}{\istermJ{\builtin{bn}{A_0, ..., A_m}{M_0, ..., M_n}}{\normalform{\subst{A_0, ..., A_n}{\alpha_0, ..., \alpha_m}{C}}}}\)}
     \end{prooftree}
-    
+
     \begin{prooftree}
-        \AxiomC{$\Gamma \vdash Q :: \typeK$}
-        \AxiomC{$Q = \mathcal{E}\{\fixT{\alpha}{S}\}$}
-        \AxiomC{$\Gamma \vdash M : \mathcal{E}\{[\fixT{\alpha}{S}/\alpha]S\}$}
-        \RightLabel{wrap}
-        \TrinaryInfC{\(\hypJ{\Gamma}{\istermJ{\wrap{\alpha}{S}{M}}{Q}}\)}
+        \AxiomC{\(\hypJ{\Gamma}{\istypeJ{A}{\typeK{}}}\)}
+        \RightLabel{error}
+        \UnaryInfC{\(\hypJ{\Gamma}{\istermJ{\error{A}}{\normalform{A}}}\)}
     \end{prooftree}
 
-Where $\mathcal{E}$ is an elimination context $\mathcal{E} ::= \bullet \mid [\mathcal{E} \,\, A]$, $A$ a type, such that $\mathcal{E}\{Q\}$ denotes the type expression obtained by replacing the $\bullet$ in $\mathcal{E}$ with $Q$. That is, $[\bullet\,\, A]\{B\} = [B\,\,A]$, and so on. 
```

Comment:
What's the status of elimination contexts, why was this deleted, etc?

Reply:
Blergh. Oversight on my part somehow. Fixed.